### PR TITLE
feat: remove & intersect with EXCLUDE & INTERSECT

### PR DESCRIPTION
**Pull request by aljazerzen**
Ref #761

### CODE REVIEW

File: prql-compiler/tests/integration/main.rs [hunk 1]
```diff
@@ -25,15 +25,33 @@ mod tests {
...
+                let opts = prql_compiler::sql::Options::default()
+                    .with_dialect(prql_compiler::sql::Dialect::PostgreSql)
+                    .no_format()
+                    .some();
+                let sql = prql_compiler::compile(&prql, opts).unwrap();
+                let pg_out = postgres::query_csv(pg_client, &sql);
+                assert_eq!(sqlite_out, pg_out);
             }
+
+            assert_snapshot!(sqlite_out);
```
Comment (max-sixty) [inline prql-compiler/tests/integration/main.rs]: Very nitty comment: to the extent we want the source expression in the insta snapshot, we can pass it with the three argument form.

File: prql-compiler/src/semantic/lowering.rs [hunk 2]
```diff
@@ -558,8 +559,40 @@ impl Lowerer {
...
+                    match self.node_mapping.get(target_id) {
+                        Some(LoweredTarget::Compute(cid)) => targets.push(*cid),
+                        Some(LoweredTarget::Input(input_columns)) => {
+                            targets.extend(input_columns.values().map(|(c, _)| c))
+                        }
+                        _ => {}
+                    }
+                }
+
+                // this is terrible code
```
Comment (max-sixty) [inline prql-compiler/src/semantic/lowering.rs:575]: lol, actually underrated comment — we should have more of these! It's also a nice way for someone to make a very small rust contribution!

File: prql-compiler/src/semantic/std.prql [hunk 3]
```diff
@@ -33,17 +33,16 @@ func aggregate<table> a<column> tbl<table> -> null
...
-)
 func group<table> by pipeline tbl<table> -> null
 func window<table> rows:0..0 range:0..0 expanding:false rolling:0 pipeline tbl<table> -> null
 
-func intersection<table> bottom<table> top<table> -> (
-    from t = _param.top | join b = _param.bottom (all (map _eq (zip t.* b.*))) | select t.*
+func append<table> `default_db.bottom`<table> top<table> -> null
+func intersect<table> `default_db.bottom`<table> top<table> -> (
+    from t = _param.top
+    join b = _param.bottom (all (map _eq (zip t.* b.*)))
```
Comment (max-sixty) [inline prql-compiler/src/semantic/std.prql:42]: What's this? :)

Comment (aljazerzen) [reply prql-compiler/src/semantic/std.prql:42]: What do you mean? the `(all (map _eq (zip t.* b.*)))` part?

`t.*` and `b.*` each evaluate to a list of columns.

`(zip t.* b.*)` evaluates to `[[t.col1, b.col1], [t.col2, b.col2], ...]`

`(map _eq (zip t.* b.*))` evaluates to `[t.col1 == b.col1, t.col2 == b.col2, ...]`

`(all (map _eq (zip t.* b.*)))` evaluates to `(t.col1 == b.col1 and t.col2 == b.col2) and ...`

... it's a join condition over all columns of both tables, abusing our resolver.

The helper functions can remain an implementation details. It'd be nice if we could hide them from the public interface.

File: prql-compiler/tests/integration/main.rs (hunk 1 above)
Comment (aljazerzen) [reply prql-compiler/tests/integration/main.rs]: I tried that, but the macro understood first argument as snapshot name.

**Pull request merged**